.github/workflows/pr-realm-js.yml

```diff
@@ -698,11 +698,32 @@ jobs:
             ~/.android/adb*
           key: avd-29
 
+      - name: Bundle test app
+        working-directory: integration-tests/environments/react-native-test-app
+        run: |
+          mkdir dist
+          npx react-native bundle --entry-file index.js --platform android --dev false --minify false --bundle-output dist/main.android.jsbundle --assets-dest dist/res
+
+      - name: Generate a keystore for signing
+        working-directory: integration-tests/environments/react-native-test-app
+        env:
+          STORE_PATH: debug.keystore
+          STORE_PASS: android
+          KEY_ALIAS: debug-key
+          KEY_PASS: android
+        run: |
+          keytool -genkey -v -keyalg RSA -keysize 2048 -validity 10000 -dname cn=Realm -keystore ${{ env.STORE_PATH }} -storepass ${{ env.STORE_PASS }} -alias ${{ env.KEY_ALIAS }} -keypass ${{ env.KEY_PASS }}
+          jq --arg storeFile ${{ env.STORE_PATH }} --arg storePassword ${{ env.STORE_PASS }} --arg keyAlias ${{ env.KEY_ALIAS }} --arg keyPassword ${{ env.KEY_PASS }} '. +{android: {signingConfigs: { release: { $storeFile, $storePassword, $keyAlias, $keyPassword } }}}' app.json > patched-app.json
+          mv patched-app.json app.json
+
       - name: Run tests
         env:
-          PLATFORM: android
+          # Limit architecture to speed up the build
+          ORG_GRADLE_PROJECT_reactNativeArchitectures: x86
+          # Enabling new architecture and bridgeless
+          ORG_GRADLE_PROJECT_newArchEnabled: true
+          ORG_GRADLE_PROJECT_bridgelessEnabled: true
```

Comment:
Our native module supports this only on Android at the moment, while https://github.com/realm/realm-js/pull/6737 will bring this to iOS too (and refactor the Android implementation too).